### PR TITLE
Fix compilation with VS 16.7

### LINKED ISSUE #704: VS2019 16.7.0 build failure
I encounter new problem building msquic.kernel.sln possibly after I upgrade VS2019 to 16.7.0.

msquic\src\platform\datapath_winkernel.c(2202): error C6011: Dereferencing NULL pointer 'Binding'. 
msquic\src\platform\hashtable.c(731): error C6387: 'ContextPtr->ChainHead' could be '0':  this does not adhere to the specification for the function 'QuicListIsEmpty'.

### PR DESCRIPTION
**Pull request by ThadHouse**
Closes #704

Found 1 actual use after free bug, 2 potential things we probably should be debug asserting, and then a few uninitialized variables that very much seem like false positives.

### CODE REVIEW

File: src/core/lookup.c [hunk 1]
```diff
@@ -55,7 +55,10 @@ QuicLookupUninitialize(
         for (uint8_t i = 0; i < Lookup->PartitionCount; i++) {
             QUIC_PARTITIONED_HASHTABLE* Table = &Lookup->HASH.Tables[i];
             QUIC_DBG_ASSERT(Table->Table.NumEntries == 0);
+#pragma warning(push)
+#pragma warning(disable:6001)
```
Comment (nibanks) [inline src/core/lookup.c:59]: Instead of pragma ignores, I'd prefer DBG asserts. What is the error for?

File: src/platform/datapath_winkernel.c [hunk 2]
```diff
@@ -2195,10 +2195,11 @@ QuicDataPathBindingReturnRecvDatagrams(
         QuicPoolFree(BatchedInternalContext->OwningPool, BatchedInternalContext);
     }
 
-    if (DataIndications != NULL) {
+    if (DataIndications != NULL && Binding != NULL) {
         //
         // Return the datagram indications back to Wsk.
         //
+        QUIC_DBG_ASSERT(Binding != NULL);
```
Comment (nibanks) [inline src/platform/datapath_winkernel.c:2202]: Assert is unnecessary with the check above.

Comment (nibanks) [reply src/platform/datapath_winkernel.c:2202]: Actually, remove the if check instead. Keep the assert.

File: src/core/lookup.c (hunk 1 above)
Comment (ThadHouse) [reply src/core/lookup.c:59]: Uninitialized memory. So can't really assert it.

File: src/platform/datapath_winkernel.c (hunk 2 above)
Comment (ThadHouse) [reply src/platform/datapath_winkernel.c:2202]: Ah thought I had done that. Whoops. Easy fix.

File: src/core/lookup.c (hunk 1 above)
Comment (ThadHouse) [reply src/core/lookup.c:59]: It's assuming that Directory in the union in QUIC_HASHTABLE  is uninitialized.

**Pull request merged**